pwm.c

```diff
@@ -17,22 +17,38 @@ pwm_scale_level(unsigned unscaled_pwm)
 }
 
 static inline int
-set_pwm_safe(furnace_context_t *ctx, unsigned new_pwm)
+set_pwm_safe(unsigned pin, furnace_context_t *ctx, unsigned new_pwm)
 {
   if (new_pwm > MAX_PWM)
     return 1;
 
   if (((unsigned) ((uint16_t) new_pwm)) != new_pwm)
     return 1;
 
-  ctx->pwm_level = new_pwm;
+  switch(pin){
+#if CONFIG_WATER
+    case WATER_PIN:
+      ctx->pwm_water = new_pwm - WATER_OFFSET;
```

Comment:
If we want `pwm_water` to be in range `(WATER_OFFSET: WATER_OFFSET+10)`, but want user to input value in range `(0: 10)`, we should do the sanitization and bias add as soon as we parse the user input and store already bias value in memory. This way we can get rid of this ugly logic of adding and subtracting `WATER_OFFSET` all over the place.

Reply:
Should we check if WATER_OFFSET + arg > MAX_PWM while getting user input ? We're already doing it in set_pwm_safe function.

Reply:
Not for `WATER_OFFSET + arg > MAX_PWM`, but for `arg > 10`